docs/site/HasMany-relation.md

```diff
@@ -0,0 +1,161 @@
+---
+lang: en
+title: 'Relations'
+keywords: LoopBack 4.0, LoopBack 4
+tags:
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/HasMany-relation.html
+summary:
+---
+
+## Defining a HasMany Relation
+
+Model relations in LoopBack 4 are defined by having a decorator on a designated
+relational property. This section describes how to define a `hasMany` relation
+at the model level using the `hasMany` decorator. The relation constrains the
+target repository by the foreign key property on its associated model.
+
+The definition of the `hasMany` relation is inferred by using the `hasMany`
+decorator. The decorator takes in the target model class constructor and
+optionally a custom foreign key to store the relation metadata. The decorator
+logic also designates the relation type and tries to infer the foreign key on
+the target model (or `keyTo` in the relation metadata) default value (source
+model name appended with `id` in camel case, also used in LoopBack 3). It also
+calls `property.array()` to ensure that the type of the property is inferred
+properly as an array of the target model instances.
+
+The decorated property name is used as the relation name and stored as part of
+the source model definition's relation metadata. The following example shows how
+to define a `hasMany` relation on a source model `Customer`.
+
+{% include code-caption.html content="/src/models/customer.model.ts" %}
+
+```ts
+import {Order} from './order.model.ts';
+import {Entity, property, hasmany} from '@loopback/repository';
+
+export class Customer extends Entity {
+  @property({
+    type: 'number',
+    id: true,
+  })
+  id: number;
+
+  @property({
+    type: 'string',
+  })
+  name: string;
+
+  @hasMany(Order) orders: Order[];
+}
+```
+
+The metadata generated for the above decoration is as follows:
+
+```js
+{
+  type: 2; // denotes hasMany relation (this is an enum value)
```

Comment:
Hmm, this is not great - the code is more difficult to debug and even more difficult to write in JavaScript (in the future). We should switch to string-based enums, so that `type` can be `hasMany`.

If you prefer, then I am ok to leave this change out of DP3 as long as you create a new GH issue labelled as `LB4 GA` and `p1`.

Reply:
Done. The change was easy enough :-).